### PR TITLE
Feat/support erc20 burn

### PR DESCRIPTION
**Pull request by DiyiWi**

### CODE REVIEW

File: internal/syncer/asset.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	contracts "github.com/meshplus/bitxhub-core/eth-contracts"
+	"github.com/meshplus/bitxhub-model/pb"
+)
+
+func (c *Client) listenBurn() {
+	for {
+		select {
+		case log := <-c.logCh:
+			// query this block from ethereum and generate mintEvent and proof for pier
+			if err := retry.Retry(func(attempt uint) error {
```
Comment (jzhe886) [inline internal/syncer/asset.go:19]: why retry it

File: go.mod [hunk 2]
```diff
@@ -38,3 +38,7 @@ require (
 )
 
 replace github.com/libp2p/go-libp2p-core => github.com/libp2p/go-libp2p-core v0.5.6
+
+replace github.com/meshplus/bitxhub-model => /Users/windiyi/GolandProjects/bitxhub-model
+
+replace github.com/meshplus/go-bitxhub-client => /Users/windiyi/GolandProjects/go-bitxhub-client
```
Comment (jzhe886) [inline go.mod]: not replace local file system

File: internal/repo/config.go [hunk 3]
```diff
@@ -123,8 +124,9 @@ func DefaultConfig() *Config {
 		Mode: Mode{
 			Type: "relay",
 			Relay: Relay{
-				Addrs:  []string{"localhost:60011"},
-				Quorum: 2,
+				Addrs:       []string{"localhost:60011"},
+				JsonrpcAddr: "ws://localhost:60011",
```
Comment (jzhe886) [inline internal/repo/config.go:128]: websocket?

File: internal/syncer/jsonrpc_client.go [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+				Receipt:    event.Recipient.String(),
+				Amount:     event.Amount.Bytes(),
+				RelayIndex: event.RelayIndex.Uint64(),
+				TxId:       event.Raw.TxHash.String(),
+			}
+			return nil
+		}, strategy.Wait(1*time.Second)); err != nil {
+			logger.Error("Can't retrieve mint event from receipt", "error", err.Error())
+		}
+	}
```
Comment (jzhe886) [inline internal/syncer/jsonrpc_client.go:73]: why retry it?

File: internal/executor/interface.go [hunk 5]
```diff
@@ -32,4 +32,15 @@ type Executor interface {
...
+	// SendBurnEvent send unlock event to plugins
+	SendBurnEvent(unLockEvt *pb.UnLock) error
+
+	QueryFilterLockStart(appchainIndex int64) int64
+
+	QueryLockEventByIndex(index int64) *pb.LockEvent
+
+	QueryAppchainIndex() int64
+
+	QueryRelayIndex() int64
```
Comment (jzhe886) [inline internal/executor/interface.go:45]: interface need more code comments

**Review by jzhe886 — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (DiyiWi) [reply go.mod]: has deal

File: internal/executor/interface.go (hunk 5 above)
Comment (DiyiWi) [reply internal/executor/interface.go]: has deal

File: internal/syncer/asset.go (hunk 1 above)
Comment (DiyiWi) [reply internal/syncer/asset.go]: has deal

File: internal/syncer/jsonrpc_client.go [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+	interchainSwapSession := &contracts.InterchainSwapSession{
+		Contract: interchainSwapContract,
+		CallOpts: bind.CallOpts{
+			Pending: false,
+		},
+	}
+
+	c := &Client{}
+	relayIndex, _ := interchainSwapSession.RelayIndex()
```
Comment (jzhe886) [inline internal/syncer/jsonrpc_client.go:120]: ignore err?

File: internal/syncer/jsonrpc_client.go [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+	c.relayIndex = relayIndex.Int64()
+	appchainIndex, _ := interchainSwapSession.AppchainIndex()
```
Comment (jzhe886) [inline internal/syncer/jsonrpc_client.go:122]: ignore err?

**Pull request closed without merging**